lib/modem_info/modem_info_json.c

```diff
@@ -141,6 +146,10 @@ int modem_info_json_string_get(char *buf)
 		}
 	}
 
+	if (CONFIG_MODEM_INFO_ADD_BOARD) {
```

Comment:
This won't work if the option is off, since the define won't be generated.
Use `IS_ENABLED(CONFIG_MODEM_INFO_ADD_BOARD)` instead.

Reply:
Fixed.